### PR TITLE
feat: error resilient atmoic ops and add a single test file

### PR DESCRIPTION
**Pull request by HarshitaKalani**
Towards: #5036

### CODE REVIEW

File: tests/errors/atomic_01_cc.f90 [hunk 1]
```diff
@@ -0,0 +1,7 @@
+program atomic_01_cc
```
Comment (Pranavchiku) [inline tests/errors/atomic_01_cc.f90:1]: Let's lump this as well

Comment (HarshitaKalani) [reply tests/errors/atomic_01_cc.f90:1]: I faced some issue with the use_fortran_env, so added this separately. Maybe, I'll look for that later if that's fine?

**Comment by gxyd** [comment]
Just a general comment: we might want to delete the stale reference files as well (e.g. reference files of `tests/errors/arithmetic_if1_cc.f90`), which wouldn't be needed anymore. If you are already aware of that, that's all good I guess.

**Comment by certik** [comment]
I think that looks great, exactly what we should do.

**Comment by HarshitaKalani** [comment]
Thanks @gxyd for pointing it out. I forgot to. I'll just push that.

**Review by certik — APPROVED** [review]
I think this looks great to me.

**Review by parth121101 — APPROVED** [review]

**Pull request merged**